openpgp/keys.go

```diff
@@ -255,7 +255,10 @@ func (el EntityList) KeysByIdUsage(id uint64, requiredUsage byte) (keys []Key) {
 				usage |= packet.KeyFlagEncryptStorage
 			} else if key.PublicKey.PubKeyAlgo == packet.PubKeyAlgoDSA {
 				usage |= packet.KeyFlagSign
+			} else if key.Entity.PrimaryKey.KeyId == id {
+				usage = (packet.KeyFlagCertify | packet.KeyFlagSign | packet.KeyFlagEncryptCommunications | packet.KeyFlagEncryptStorage)
```

Comment:
good idea, done!

On Tue, Jul 12, 2016 at 1:02 AM, oconnor663 <notifications@github.com>
wrote:

> In openpgp/keys.go
> <https://github.com/keybase/go-crypto/pull/35#discussion_r70377495>:
>
> > @@ -255,7 +255,10 @@ func (el EntityList) KeysByIdUsage(id uint64, requiredUsage byte) (keys []Key) {
> >  				usage |= packet.KeyFlagEncryptStorage
> >  			} else if key.PublicKey.PubKeyAlgo == packet.PubKeyAlgoDSA {
> >  				usage |= packet.KeyFlagSign
> > +			} else if key.Entity.PrimaryKey.KeyId == id {
> > +				usage = (packet.KeyFlagCertify | packet.KeyFlagSign | packet.KeyFlagEncryptCommunications | packet.KeyFlagEncryptStorage)
>
> Would another possible fix be to add an RSA-specific block here, similar
> to what we have above with DSA and ElGamal? (Or maybe, one for each of the
> 3 different usage-specific RSA aliases that rfc4480 defines for G-d only
> knows what reason?) I'm a little worried about this branch deciding that a
> key is usable for encryption when it really isn't.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/go-crypto/pull/35/files/1812edc25c97d65cfb76dfbb3d40607fc1ef73ac#r70377495>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA05_6Y9N-aAZVMW-47mwQyRjKnk9bcIks5qUyADgaJpZM4JJ9Mj>
> .
>